### PR TITLE
Fixed AttributeBehavior and TimestampBehaviour based on new Event system

### PR DESCRIPTION
**Pull request by FabrizioCaldarelli**
Needed to pass db-pgsql package tests

### CODE REVIEW

File: src/behaviors/TimestampBehavior.php [hunk 1]
```diff
@@ -93,14 +94,12 @@ class TimestampBehavior extends AttributeBehavior
...
+    public function __construct()
     {
-        parent::init();
-
         if (empty($this->attributes)) {
             $this->attributes = [
-                BaseActiveRecord::EVENT_BEFORE_INSERT => [$this->createdAtAttribute, $this->updatedAtAttribute],
-                BaseActiveRecord::EVENT_BEFORE_UPDATE => $this->updatedAtAttribute,
+                ActiveRecordSaveEvent::BEFORE_INSERT => [$this->createdAtAttribute, $this->updatedAtAttribute],
+                ActiveRecordSaveEvent::BEFORE_UPDATE => $this->updatedAtAttribute,
```
Comment (rob006) [inline src/behaviors/TimestampBehavior.php]: This constructor does not make much sense now - unless you extend this class, you will not be able to configure `attributes`, `createdAtAttribute` nor `updatedAtAttribute` properties before this code is executed. And after constructor was called configuring `createdAtAttribute` and `updatedAtAttribute` does not have any effect - these properties does not make any sense now, they're should be constructor arguments (or implement config array as constructor argument).

Comment (FabrizioCaldarelli) [reply src/behaviors/TimestampBehavior.php]: We can use TimestampBehaviour from behaviors() method:

```
public function behaviors()
{
    return [
        TimestampBehavior::className(),
    ];
}
```

How could you pass parameters?

**Comment by rob006** [comment]
```php
public function behaviors() {
    return [
        new TimestampBehavior(/* arguments */),
    ];
}
```

Or using array syntax supported by `Yii::createObject()`.

File: src/behaviors/TimestampBehavior.php (hunk 1 above)
Comment (rob006) [reply src/behaviors/TimestampBehavior.php]:
```php
public function behaviors() {
    return [
        new TimestampBehavior(/* arguments */),
    ];
}
```

Or using array syntax supported by `Yii::createObject()`.

Comment (FabrizioCaldarelli) [reply src/behaviors/TimestampBehavior.php]: I think that this should work:

```php
public function behaviors() {
    return [
        Yii::createObject(['__class' => TimestampBehavior::class, 'created_at' => '...', 'updated_at' => '...')
    ];
}
```

Comment (FabrizioCaldarelli) [reply src/behaviors/TimestampBehavior.php]: Anyway, I'm agree with defining TimestampBehavior.

File: src/behaviors/TimestampBehavior.php [hunk 2]
```diff
@@ -138,4 +141,14 @@ public function touch($attribute)
         }
         $owner->updateAttributes(array_fill_keys((array) $attribute, $this->getValue(null)));
     }
+
+    private function getCreatedAtAttribute() : string
```
Comment (rob006) [inline src/behaviors/TimestampBehavior.php]: Why do you need private getter for private property? :D

Comment (FabrizioCaldarelli) [reply src/behaviors/TimestampBehavior.php]: Guess ... :D

Comment (FabrizioCaldarelli) [reply src/behaviors/TimestampBehavior.php]: Fixed

**Comment by samdark** [comment]
Merged. Thanks!

**Pull request merged**